### DISCUSSION TITLE
highlight the elements

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3012] [Q&A]
Hello

how can I simulate clicking on an element to make the element selected?

I do a little prologue
I use Ecsql to determine the userlabel present in my model

SELECT UserLabel, count (*) ElementCount FROM bis.SpatialElement GROUP BY UserLabel ORDER BY UserLabel

I insert these elements in a summary list (<Select>) after which, selecting the list I would like to highlight the elements that are part of that specific UserLabel

I can add the elements through the selectionSet.elements.add () but the element is not highlighted
even if present among the elements

I tried to use emphasizeElements too but it's not what I want or I'm using it wrong ..

Thanks for your help

Paul

**Comment by pmconne** [comment]
Hi @PaoloPetra,

The most likely explanation is that the elements returned by your query do not have geometry of their own but are instead part of an [assembly](https://www.itwinjs.org/bis/intro/element-fundamentals/#assemblies) wherein child elements contain the geometry.
The iTwin.js [presentation packages](https://www.itwinjs.org/learning/presentation/) use the concept of "selection scope" to define what gets selected when an element is added to the selection set. One option is "assembly", which is probably relevant here.
Which application template are you using? Or, if none, are you using the presentation libraries?

**Reply by pmconne** [reply]
[This sample](https://www.itwinjs.org/sample-showcase/?group=AppUI&sample=viewport-frontstage-sample&imodel=Bay+Town+Process+Plant) demonstrates how selection scope works. Note the "Scope" combo box in the bottom-right corner.
If I set it to "Element" and select this portion of a pipe, only the element representing that portion will be highlighted:
![image](https://user-images.githubusercontent.com/22944042/148939992-eec67100-18cf-4f1e-bbee-962659542d1d.png)
If I set it to "Top Assembly" and select any portion of the same pipe, all of the individual elements that make up the pipe assembly will be highlighted:
![image](https://user-images.githubusercontent.com/22944042/148940112-2c858dea-65fa-4221-b6aa-17ea369b6eca.png)
